frontend/views/containers/GroupPledgesGraph.vue

```diff
@@ -1,218 +1,195 @@
-      }
-    }
-  },
-  mounted: function () {
-    window.addEventListener('resize', this.handleResize)
-  },
-  beforeDestroy: function () {
-    window.removeEventListener('resize', this.handleResize)
-  },
   computed: {
-    chartSize () {
-      return this.isMobile ? '8rem' : undefined
-    },
-    legendItemClass () {
-      return 'column c-legend-item'
-    },
+    ...mapGetters([
+      'groupSettings',
+      'groupProfiles',
+      'groupMembersCount',
+      'ourContractId'
+    ]),
     graphData () {
-      const { userPledgeAmount, userIncomeAmount } = this
-      const { groupMembersTotal, groupPledgeGoal, mincome, othersPledges } = this.fakeStore
-      const othersAmount = othersPledges.reduce((acc, cur) => acc + cur, 0)
-      const userIncomeNeeded = userIncomeAmount !== null && userIncomeAmount < mincome ? mincome - userIncomeAmount : null
-      const totalAmount = othersAmount + userPledgeAmount
-      const goal = groupPledgeGoal + userIncomeNeeded
-      const members = othersPledges.length + (userPledgeAmount ? 1 : 0)
-      const neededPledges = goal - totalAmount
-      const surplus = totalAmount - groupPledgeGoal
-      const membersNeedingPledges = groupMembersTotal - othersPledges.length
-
-      // Dumb Rule of 3 algorithm to know how much the user will receive (userIncomeToReceive) - just for layout purposes.
-      const avgMembersNeedingPledges = goal / membersNeedingPledges
-      const avgNeededPledges = neededPledges / membersNeedingPledges
-      const userIncomeToReceive = userIncomeNeeded ? (userIncomeNeeded - (userIncomeNeeded * avgNeededPledges / avgMembersNeedingPledges)).toFixed(2) : 0
+      const mincome = this.groupSettings.mincomeAmount
+      const ourPledgeAmount = this.type === 'pledgeAmount' && this.amount >= 0 && this.amount
+      const ourIncomeAmount = this.type === 'incomeAmount' && this.amount < mincome && this.amount
+      const doWeNeedPledge = typeof ourIncomeAmount === 'number'
+      let othersIncomeNeeded = 0
+      let othersPledgesAmount = 0
+      let othersPledgesCount = 0
+
+      Object.keys(this.groupProfiles).forEach(username => {
+        const { incomeDetailsType, contractID, ...profile } = this.groupProfiles[username]
+
+        if (contractID === this.ourContractId) { return }
+
+        const amount = profile[incomeDetailsType]
+
+        if (incomeDetailsType === 'incomeAmount') {
+          othersIncomeNeeded += mincome - amount
+        } else if (incomeDetailsType === 'pledgeAmount') {
+          othersPledgesAmount += amount
+          othersPledgesCount += 1
+        }
+      })
+
+      const ourIncomeNeeded = doWeNeedPledge ? mincome - ourIncomeAmount : null
+      const pledgeTotal = othersPledgesAmount + ourPledgeAmount
+      const groupGoal = othersIncomeNeeded + ourIncomeNeeded
+      const neededPledges = groupGoal - pledgeTotal
+      const surplus = pledgeTotal - othersIncomeNeeded - ourIncomeNeeded
+
+      let ourIncomeToReceive
```

Comment:
This can be initialized to `ourIncomeNeeded` so that lines 110-112 can be removed